server/src/executables.ts

```diff
@@ -0,0 +1,91 @@
+import * as Fs from 'fs'
+import * as Path from 'path'
+
+import * as ArrayUtil from './util/array'
+import * as FsUtil from './util/fs'
+import * as ShUtil from './util/sh'
+
+/**
+ * Provides information based on the programs on your PATH
+ */
+export class Executables {
+  /**
+   * @param path is expected to to be a ':' separated list of paths.
+   */
+  public static fromPath(path: string): Promise<Executables> {
+    const paths = path.split(':')
+    const promises = paths.map(x => _executables(x))
+    return Promise.all(promises)
+      .then(ArrayUtil.flatten)
+      .then(ArrayUtil.uniq)
+      .then(executables => new Executables(executables))
+  }
+
+  private executables: Set<string>
+
+  private constructor(executables: string[]) {
+    this.executables = new Set(executables)
+  }
+
+  /**
+   * Find all programs in your PATH
+   */
+  public list(): Array<string> {
+    return Array.from(this.executables.values())
+  }
+
+  /**
+   * Check if the the given {{executable}} exists on the PATH
+   */
+  public isExecutableOnPATH(executable: string): boolean {
+    return this.executables.has(executable)
+  }
+
+  /**
+   * Look up documentation for the given executable.
+   *
+   * For now it simply tries to look up the MAN documentation.
+   */
+  public documentation(executable: string): Promise<string> {
+    return ShUtil.exec(`man ${executable} | col -b`).then(doc => {
+      return !doc
+        ? Promise.resolve(`No MAN page for ${executable}`)
+        : Promise.resolve(doc)
+    })
+  }
+}
+
+/**
+ * Find all the executables on the given path.
+ * Only returns direct children, or the path itself if it's an executable.
+ */
+function _executables(path: string): Promise<string[]> {
```

Comment:
We would call this `findExecutablesInPath`, then we can almost get rid of the code comment.